concepts/pointers/introduction.md

```diff
@@ -0,0 +1,131 @@
+# Introduction
+
+Like many other languages, C++ has _pointers_.
+You already know _references_ and _pointers_ are similar, but think of them as a level closer to the inner workings of your computer.
+_Pointers_ are variables that hold object addresses.
+They are used to directly interact with objects, enabling dynamic memory allocation and efficient data manipulation in C++.
+
+If you're new to _pointers_, they can feel a little mysterious but once you get used to them, they're quite straight-forward.
+
+They're a crucial part of C++, so take some time to really understand them.
+The bare-bone version in this concept is also called _dumb pointer_ or _raw pointer_.
+With modern C++ there are also _smart pointers_, the basic type is not smart at all and you have to handle all the work manually.
+
+Before digging into the details, it's worth understanding the use of _pointers_.
+_Pointers_ are a way to share an object's address with other parts of our program, which is useful for two major reasons:
+1. Like _references_, pointers avoid copies and help to reduce the resource-footprint of your program.
+2. Unlike _references_, pointers can be reassigned to different objects.
+3. Pointers can also point to a null value, to indicate, that they currently do not point to any object.
+
+## General Syntax
+
+A pointer declaration in C++ involves specifying the data type to which the the pointer is pointing, followed by an asterisk (`*`) and the pointer's name.
+When pointers are declared, they are not automatically initialized.
+Without explicit assignment, a pointer typically holds an indeterminate value, often referred to as a "garbage address."
+While certain compilers might initialize pointers to `nullptr`, this behavior is not guaranteed across all compilers, so it's essential not to rely on it.
+It's best practice to explicitly initialize raw pointers and verify their non-null status before utilization to avoid potential issues.
+
+```cpp
+int* ptr{nullptr}; // Declares a pointer and makes sure it is not invalid
+```
+
+To assign the address of a variable to a pointer, you use the address-of operator (`&`).
+Dereferencing a pointer is done using the _indirection operator_ (`*`) operator.
+
+```cpp
+std::string opponent{"Solomon Lane"};
+// 'ethan' points to the address of the string opponent
+std::string* ethan{&opponent}; 
+// Instead of ethan's, the opponent's name address is given to the passPort
+std::string passportName{*ethan};
+```
+
+Attention: dereferencing has to be done explicitly, while _references_ just worked like an alias.
+
+## Pointer Arithmetic
+
+_Pointer arithmetic_ allows you to perform arithmetic operations on pointers, which is particularly useful when working with arrays.
+Adding an integer to a pointer makes it point to a different element.
+
+```cpp
+// Stargate addresses
```

Comment:
Since this concept is about pointers and addresses using the term "stargate addresses" might be a little bit confusing (one is the technical term related to the concept, the other is the fictional term for interstellar travel.)